### PR TITLE
[improvement](publish) fix publish timeout in cocurrent load

### PR DESCRIPTION
**Pull request by yixiutt**
In concurrent load, some publish timeout happens occasionally. This is cause by meta lock hold by other thread so publish add increase rowset hang for several seconds.

StorageEngine::start_delete_unused_rowset will hold gc_mutex and it cost a lot of time, so that add_used_rowset wait lock, and compaction modify_rowset or other tablet method will hold meta_lock and call add_unused_rowset which will make meta_lock occupied for too long, finally makes publish timeout.

In this pr, I copy unused_rowsets in lock and delete these rowset without lock, makes gc_mutex more lightweight so meta lock can be acquired immediately in publish thread. 

Also, i modify tablet_rowset_stale_sweep_time_sec to 300 cause too much stable rowset will make tablet_meta too big when save, save_meta will hold meta_lock and than block publish thread

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.35 seconds
 load time: 442 seconds
 storage size: 17153697998 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221113153556_clickbench_pr_45209.html

File: be/src/olap/storage_engine.cpp [hunk 1]
```diff
@@ -862,11 +862,13 @@ void StorageEngine::_parse_default_rowset_type() {
...
-        if (it->second.use_count() != 1) {
-            ++it;
-        } else if (it->second->need_delete_file()) {
+    std::unordered_map<std::string, RowsetSharedPtr> unused_rowsets_copy;
+    {
+        std::lock_guard<std::mutex> lock(_gc_mutex);
+        unused_rowsets_copy = _unused_rowsets;
+    }
+    for (auto it = unused_rowsets_copy.begin(); it != unused_rowsets_copy.end(); ++it) {
+        if (it->second.use_count() == 1 && it->second->need_delete_file()) {
```
Comment (dataroaring) [inline be/src/olap/storage_engine.cpp:871]: I am not sure that use_count is at least 2 after 'unused_rowsets_copy = _unused_rowsets'.

**Review by dataroaring — DISMISSED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by zhannngchen — APPROVED** [review]
LGTM

**Pull request merged**